docs/docs/setup/web-wasm-setup.md

```diff
@@ -1,30 +1,103 @@
 
-    ```bash
-    test-e2e/web $ yarn && yarn start
-    ```
+### 1. Modify 'Cargo.toml' in the "mopro-wasm-lib"
 
-3. **Verify the Results**:  
-   Open a web browser and visit the test page at the default url: `http://localhost:3000`.
-   
-   Check the results displayed in the browser console or user interface.
+The user-defined circuit implmentation crate should be added as a dependency manually, as illustrated below:
+
+```rust
+[package]
+name = "mopro-wasm-lib"
+version = "0.1.0"
+edition = "2021"
+
+[lib]
+crate-type = ["rlib", "cdylib"]
+
+[dependencies]
+// mopro-wasm = { git = "https://github.com/zkmopro/mopro",features = [
+//     "gemini",
+//     "hyperplonk",
+//     "plonk",
+// ]}
+my-halo2-circuit = { git = "http://github.com/users/my-halo2-circuit.git" }
+
+[target.wasm32-unknown-unknown.dependencies]
+console_error_panic_hook = "0.1.7"
+getrandom = { version = "0.2.15", features = ["js"] }
+serde-wasm-bindgen = "0.6.5"
+wasm-bindgen = { version = "0.2.95", features = ["serde-serialize"] }
+wasm-bindgen-console-logger = "0.1.1"
+wasm-bindgen-futures = "0.4.47"
+wasm-bindgen-rayon = { version = "1.2.2", features = ["no-bundler"] }
+wasm-bindgen-test = "0.3.42"
+
+```
+
+The `mopro-wasm` crate no longer required for compiling users-defined circuit implementation: **"my-halo2-circuit"** in that case.
+
+### 2. Create Wrapper Functions for Generate/Verify proof method
+
+To compile Wasm code with the circuit, wrapper functions for generating and verifying proof methods in the user-defined circuit implementation must be created in `mopro-wasm-lib/src/lib.rs`, using the example structure provided below:
+
+```rust
+use my_halo2_circuit;
+
+#[wasm_bindgen]
+pub fn generate_proof(input: JsValue) -> Result<JsValue, JsValue> {
+   // function implementations with `my-halo2-circuit`
+   let proof = my_halo2_circuit::generate_proof(parsed_input);
+   to_value(...)
+}
+
+#[wasm_bindgen]
+pub fn verify_proof(proof: JsValue, public_inputs: JsValue) -> Result<JsValue, JsValue> {
+   // function implementations with `my-halo2-circuit`
+   let result = my_halo2_circuit::verify_proof(parsed_proof, parsed_public_input);
+   to_value(...)
+}
+```
+
+### 3. Build again for web
+
+To ensure a clean build, remove the existing `MoproWasmBindings` directory in the `mopro-example-app`, which was previously generated with `mopro-wasm-lib`.
+Then, execute the `mopro build` command again, selecting the "web" platform in `mopro-example-app`:
+
+```shell
+mopro-example-app $ rm -rf MoproWasmBindings
+mopro-example-app $ mopro build
+```
+
+### 4. **Integrate the Wasm Code**:
+
+The wasm code can be imported and used in a web application, as illustrated below:
+
+```javascript
+const mopro_wasm = await import('./pkg/my-halo2-circuit.js');
```

Comment:
does it change the name of the package?
I suppose the library will be called `mopro_wasm_lib.js` with `mopro build`?

Reply:
Ah, its left over. I fixed at [8693285](https://github.com/zkmopro/mopro/pull/292/commits/8693285140d906974070180700213377fc7d12d1)